alibi/saving.py

```diff
@@ -373,6 +375,27 @@ def _load_CounterfactualRLTabular(path: Union[str, os.PathLike],
     return _helper_load_CounterfactualRL(path, predictor, explainer)
 
 
+def _save_SimilarityExplainer(explainer: 'GradientSimilarity', path: Union[str, os.PathLike]) -> None:
+    predictor = explainer.predictor
+    explainer.predictor = None  # type: ignore[assignment]
+
+    with open(Path(path, 'explainer.dill'), 'wb') as f:
+        dill.dump(explainer, f, recurse=True)
+
+    explainer.predictor = predictor
+
+
+def _load_SimilarityExplainer(path: Union[str, os.PathLike],
+                              predictor: 'Union[tf.keras.Model, torch.nn.Module]',
+                              meta: dict) -> 'GradientSimilarity':
+    # load explainer
+    with open(Path(path, "explainer.dill"), "rb") as f:
+        explainer = dill.load(f)
+    explainer.reset_predictor(predictor)
+
+    return explainer
+
+
```

Comment:
What happens when the explainer precomputes the gradients? I believe that resetting the predictor should reset the precomputed gradients too. Otherwise we can potentially use the gradients of a different model. Probably have to rethink what we want to save? This is related to a previous comment about how we check if an explainer is fitted and can emphasize why we want/need to consider the precomputation flag.

Reply:
Maybe is worth adding this case in a saving test.

Reply:
Ass discussed offline, the primary reason for `rest_predictor` is to allow switching a black-box predictor to the deployment endpoint in production. For white-box explainers the method is kept for API consistency, in the future we might serialize the white-box model transparently.

In the mean-time, we trust the user to use the same model when loading (perhaps we could do something clever like save a hash of the runtime model and check it agrees with the one provided on loading?). We shouldn't clear saved gradients as the expected use of `save_explainer` is so we can just call `explain` without having to re-fit.